### PR TITLE
Teach bmac to handle reletive references, in particular -r:System.Drawing

### PR DESCRIPTION
**Pull request by chamons**
- More fixes for https://github.com/xamarin/xamarin-macios/issues/3948

### CODE REVIEW

File: src/btouch.cs [hunk 1]
```diff
@@ -648,3 +630,37 @@ static string GetWorkDir ()
...
+		}
+		if (forceSystemDrawing && !ContainsReference (references, "System.Drawing"))
+			AddSDKReference (references, sdk_path, "System.Drawing.dll");
+	}
+
+	static bool ExistsInSDK (string sdk_path, string name) => File.Exists (Path.Combine (sdk_path, name));
+
+	static void FixRelativeReferences (string sdk_path, List<string> references)
+	{
+		foreach (var r in references.Where (x => ExistsInSDK (sdk_path, x + ".dll")).ToList ()) {
```
Comment (rolfbjarne) [inline src/btouch.cs]: Why the `.ToList ()`? It looks redundant (and slower).

**Comment by alanmcgovern** [comment]
Is there any reason we can’t produce packages for PRs? It seems backwards to me to wait until things merge to master before testing them!

**Comment by rolfbjarne** [comment]
@alanmcgovern it's in progress, we'll get there soon.

**Comment by alanmcgovern** [comment]
Ah I didn’t realise! The only thing I heard was that they weren’t created for PRs. I wasn’t sure if it was a deliberate choice or just a temporary limitation :)

File: src/btouch.cs (hunk 1 above)
Comment (chamons) [reply src/btouch.cs]: I don't believe you can remove items from a list while enumerating it. Thus we make a copy of the list to enumerate while mutating the original.

Comment (chamons) [reply src/btouch.cs]: Confirmed, check https://dotnetfiddle.net/tVGT50 and compare with and without the define commented out.

```[System.InvalidOperationException: Collection was modified; enumeration operation may not execute.]```

Comment (rolfbjarne) [reply src/btouch.cs]: OK, I didn't see that you were modifying the list.

Comment (chamons) [reply src/btouch.cs]: I don't love the pattern, but i wasn't sure of a cuter way to do it.

**Pull request merged**